src/prefect/cli/cloud.py

```diff
@@ -150,13 +288,81 @@ async def login(
     Creates a new profile configured to use the specified PREFECT_API_KEY.
     Uses a previously configured profile if it exists.
     """
+    if not app.console.is_interactive and (not key or not workspace_handle):
+        exit_with_error(
+            "When not using an interactive terminal, you must supply a `--key` and `--workspace`."
+        )
+
+    profiles = load_profiles()
+    current_profile = get_settings_context().profile
+
+    if key and PREFECT_API_KEY.value() == key:
+        exit_with_success("This profile is already authenticated with that key.")
+
+    already_logged_in_profiles = []
+    for name, profile in profiles.items():
+        profile_key = profile.settings.get(PREFECT_API_KEY)
+        if (
+            not key
+            and profile_key is not None
+            or (key and profile_key == key)
+            and await check_key_is_valid_for_login(profile_key)
+        ):
+            already_logged_in_profiles.append(name)
+
+    current_profile_is_logged_in = current_profile.name in already_logged_in_profiles
+
+    if current_profile_is_logged_in:
+        app.console.print("It looks like you're already authenticated on this profile.")
+        should_reauth = typer.confirm(
+            "? Would you like to reauthenticate?", default=False
+        )
+        if not should_reauth:
+            app.console.print("Using the existing authentication on this profile.")
+            key = PREFECT_API_KEY.value()
+
+    elif already_logged_in_profiles:
+        app.console.print(
+            "It looks like you're already authenticated with another profile."
+        )
+        if not typer.confirm(
+            "? Would you like to reauthenticate with this profile?", default=False
+        ):
+            if typer.confirm(
+                "? Would you like to switch to an authenticated profile?", default=True
+            ):
+                profile_name = prompt_select_from_list(
+                    app.console,
+                    "Which authenticated profile would you like to switch to?",
+                    already_logged_in_profiles,
+                )
+
+                profiles.set_active(profile_name)
+                save_profiles(profiles)
+                exit_with_success(
+                    f"Switched to authenticated profile {profile_name!r}."
```

Comment:
```suggestion
                    f"Switched to profile {profile_name!r}."
```

Reply:
You do not think it's worth noting that the profile is authenticated?